addons/sale_pdf_quote_builder/models/ir_actions_report.py

```diff
@@ -138,8 +138,9 @@ def _get_form_fields_mapping(self, order):
         }
 
         # Adding fields from each line, prefixed by the line_id to avoid conflicts
-        for line in order.order_line:
-            form_fields_mapping.update(self._get_sol_form_fields_mapping(line))
+
+        [form_fields_mapping.update(self._get_sol_form_fields_mapping(line))
+            for line in order.order_line if line.id in doc_line_id_mapping.values()]
```

Comment:
For clarity sake, keep it as before and filter the lines in line 141 (and take off that new line, that comment is linked to it)